### PR TITLE
Improve mutable array benchmarks

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs [hunk 1]
```diff
@@ -17,23 +17,81 @@
...
+import Control.Monad.IO.Class (MonadIO)
+import Data.Functor ((<&>))
 import Streamly.Prelude (MonadAsync, SerialT, IsStream)
+import Streamly.Internal.Data.Stream.Serial (SerialT(..))
 import System.Random (randomRIO)
 
-import qualified Streamly.Prelude as Stream
 import qualified Streamly.Internal.Data.Array.Foreign as Array
 import qualified Streamly.Internal.Data.Array.Foreign.Mut as MArray
+import qualified Streamly.Internal.Data.Stream.StreamD as D
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:29]: import as `StreamD`.

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs [hunk 2]
```diff
@@ -17,23 +17,81 @@
...
+import qualified Streamly.Prelude  as S
+import qualified Streamly.Prelude as Stream
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:31]: Need only one import.

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs [hunk 3]
```diff
@@ -17,23 +17,81 @@
...
+
+import Gauge
+import Prelude as P hiding (map)
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:34]: import qualified, better import the full name Prelude rather than P.

**Review by harendra-kumar — APPROVED** [review]

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs (hunk 3 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:34]: importing this `quialified` is rather tedious to work with.

Comment (adithyaov) [reply benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs]: Doing it though.

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs [hunk 4]
```diff
@@ -89,6 +280,13 @@ main = do
...
 
     allBenchmarks array value =
-        [ bgroup (o_1_space_prefix moduleName) $
-            o_1_space_serial_marray value array
+        [ bgroup (o_1_space_prefix moduleName)
+              $ concat
+                    [ o_1_space_serial_marray value array
+                    , o_1_space_generation value
+                    , o_1_space_elimination value
+                    , o_n_heap_serial value
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:288]: Why is `o_n_heap` added under `o_1_space_prefix`? The benchmarks may fail to run when run under restricted memory.

File: benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs [hunk 5]
```diff
@@ -43,29 +122,141 @@ sourceUnfoldrM value n = Stream.unfoldrM step n
...
+        _ -> undefined
+
+{-# INLINE scanl' #-}
+{-# INLINE scanl1' #-}
+{-# INLINE map #-}
+
+scanl' , scanl1', map
+    :: MonadIO m => Int -> Int -> Stream Int -> m (Stream Int)
+
+
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs:147]: Remove extra blank line.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**